pullRequestHandler.go

```diff
@@ -38,6 +38,15 @@ func handlePullRequest(req types.PullRequestOuter) {
 
 	hasUnsignedCommits, err := hasUnsigned(req, client)
 
+	authorAssociation := req.PullRequest.AuthorAssociation
+
+	if firstTimeContributor(authorAssociation) == true {
```

Comment:
Small one, but is there a way to test whether the label is already there with the already received webhook payload?

If there is we could save an API call if we only hit `AddLabelsToIssue` where the first-timer label isn't already there.  We do something similar with `manageLabel()`

Reply:
Yes I would only expect this code to execute if this is present in the body of the request at the root: `"action": "opened"`.

Reply:
The check is run on the PR being opened, not at any other time, so we can save the call by combining the logic. The label is also never removed and the user could have multiple Pos with this label (I think that's OK) until one is merged then future PRs from that point will not get the label.

Reply:
Yeah i think this is a good idea(to have the label up till some point).If i can play around with derek on my repo i will try to reference this.